packages/pwa/app/utils/routes-utils.js

```diff
@@ -5,38 +5,114 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-// import {getUrlConfig} from './utils'
+import {getDefaultSite, getSites} from './site-utils'
 import {urlPartPositions} from '../constants'
 
 /**
- * Configure the routes based on url configuration provided.
+ * Construct literal routes based on url config
+ * with each site id/alias and locale id/alias pair from sites config
  *
  * @param {array} routes - array of routes to be configured
- * @param {object} urlConfig - the url configuration object
- * @param {object} - a custom configured object
+ * @param {object} urlConfig
+ * @param {object} options
+ * @param {array} options.ignoredRoutes - routes that does not need configured
  * @return {array} - list of configured route objects
  */
 export const configureRoutes = (routes = [], urlConfig, {ignoredRoutes = []}) => {
     if (!routes.length) return []
-    if (!urlConfig) return routes
 
-    return routes.map((route) => {
-        const {path, ...rest} = route
-        if (ignoredRoutes.includes(path)) return route
-        let basePathSegments = []
+    const allSites = getSites()
+    const defaultSite = getDefaultSite()
+    if (!allSites) return routes
 
-        const options = ['locale']
+    let outputRoutes = []
+    for (let i = 0; i < routes.length; i++) {
+        const {path, ...rest} = routes[i]
 
-        options.forEach((option) => {
-            const position = urlConfig[option]
-            if (position === urlPartPositions.PATH) {
-                basePathSegments.push(`:${option}`)
-            }
-        })
+        if (ignoredRoutes.includes(path)) {
+            outputRoutes.push(routes[i])
+        } else {
+            const showDefaults = urlConfig.showDefaults
+            const defaultSites = [defaultSite.id, defaultSite.alias]
+            allSites.forEach((site) => {
+                // append site ids and aliases to an array
+                const sites = [site.alias, site.id].filter(Boolean)
```

Comment:
Let's rename this `sites` to `siteRefs`.

Reply:
Done!